src/services/print.js

```diff
@@ -252,14 +252,16 @@ ngeo.Print.prototype.encodeImageWmsLayer_ = function(arr, layer) {
 
   var url = source.getUrl();
   var params = source.getParams();
-  this.encodeWmsLayer_(arr, layer.getOpacity(), url, params);
+  if (goog.isDefAndNotNull(url)) {
```

Comment:
It's more processing. It's confusing for readers of the code, who may think that `source.getUrl()` may return `null`. I've become strict with typing, and I think it's right :)